articles/coding-on-purpose.md

```diff
@@ -0,0 +1,99 @@
+---
+authorName: Adrian Herzog
+authorGithubUsername: adiherzog
+---
+
+# Coding On Purpose
+
+&copy; 2017 Adrian Herzog
+
+## Purpose Is Not A Modern Luxury
+
+I'm born 1986, so I'm definitely a member of the infamous *Generation Y* a.k.a. *Millenials*, which includes everybody born between roughly 1980 and 2000. Generation Y is often characterized as *Gerneration Why*: as a generation that questions everything. When it comes to jobs we supposedly are not happy with a good salary and nice colleagues, no, we want a job that serves a higher purpose. We do not just want to work, we want to make an impact, they say.
+
+No matter whether all this talk is true, one thing is clear: making an impact is an economical necessity if your business wants to survive in the long run. Or as Paul Graham of *Y Combinator* - note the Y in their name - says:
+
+> Make something people want.
+
+By the way: Justin Bieber - born 1994 - is a millenial too. And so it's no surprise that his current (2017) world tour is titled the *Purpose World Tour*.
+
+When building software, the *why* or better the *what for* question is a very important one. I'm aware that this is nothing new for you. But this is what makes it easy to forget and to lose the focus on it. We usually start our projects with a more or less clear vision of what we want to achieve - how the world will be better when we're done. Then we start building. We start setting milestones, and soon reaching the milestone becomes our highest goal. In order to meet milestones we need to deliver a certain amount of user stories per sprint (or so we think...) and soon estimating effort in story points and tracking velocity become our way of steering the project. Or we might be busy optimizing the hourly rate of our sub-contractors from another continent. Whenever our focus drifts away from achieving the original purpose, we need to remind ourselves:

```

Comment:
"more or less clear vision of what we want to achieve". 
Who is we? Software builders? If yes, does the client have the same vision? end user? What is the real why? What is the impact we want to achieve?

Idea: Maybe the 5 why root cause analysis finds the real vision?

Reply:
Now it becomes interessting! I think this paragraph is the best so far and this part should be in the introduction.

Reply:
and @peitor you do not like your own comment? :see_no_evil: :wink: